src/cryptonote_basic/cryptonote_format_utils.h

```diff
@@ -64,6 +64,8 @@ namespace cryptonote
   crypto::public_key get_tx_pub_key_from_extra(const transaction_prefix& tx, size_t pk_index = 0);
   crypto::public_key get_tx_pub_key_from_extra(const transaction& tx, size_t pk_index = 0);
   bool add_tx_pub_key_to_extra(transaction& tx, const crypto::public_key& tx_pub_key);
+  bool add_tx_pub_key_to_extra(transaction_prefix& tx, const crypto::public_key& tx_pub_key);
```

Comment:
Since `transaction` is derived from `transaction_prefix`, this overload can replace the existing function with `transaction& tx`.